From aa4d663c964c40b772613f2aca419c6e3d2a5a64 Mon Sep 17 00:00:00 2001
From: Shubhrajyoti Datta <shubhrajyoti.datta@xilinx.com>
Date: Mon, 6 Jul 2020 15:51:10 +0530
Subject: [PATCH 1425/1851] misc: trafgen: fix out of scope warning

commit fb828254a5900326571748aba35f2fd18667b31d from
https://github.com/Xilinx/linux-xlnx.git

In xtg_pram_write the param_word is copied to data
and data is used after the scope of param_word. Fix the same
by moving the declaration of param_word. While at it more the
declaration of other variable to the start of the function.

Signed-off-by: Shubhrajyoti Datta <shubhrajyoti.datta@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/misc/xilinx_trafgen.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/misc/xilinx_trafgen.c b/drivers/misc/xilinx_trafgen.c
index 89187c0c50f0..0c0c5c40874f 100644
--- a/drivers/misc/xilinx_trafgen.c
+++ b/drivers/misc/xilinx_trafgen.c
@@ -1187,6 +1187,8 @@ static ssize_t xtg_pram_write(struct file *filp, struct kobject *kobj,
 	struct xtg_dev_info *tg =
 		to_xtg_dev_info(container_of(kobj, struct device, kobj));
 	u32 *data = (u32 *)buf;
+	struct xtg_pram *cmdp = (struct xtg_pram *)buf;
+	u32 param_word;
 
 	if (off >= XTG_PARAM_RAM_SIZE) {
 		pr_err("Requested Write len exceeds 2K PRAM size\n");
@@ -1198,8 +1200,6 @@ static ssize_t xtg_pram_write(struct file *filp, struct kobject *kobj,
 
 	/* Program each command */
 	if (count == sizeof(struct xtg_pram)) {
-		struct xtg_pram *cmdp = (struct xtg_pram *)buf;
-		u32 param_word;
 
 		if (!cmdp)
 			return -EINVAL;
@@ -1278,6 +1278,8 @@ static ssize_t xtg_cram_write(struct file *filp, struct kobject *kobj,
 	struct xtg_dev_info *tg =
 		to_xtg_dev_info(container_of(kobj, struct device, kobj));
 	u32 *data = (u32 *)buf;
+	struct xtg_cram *cmdp = (struct xtg_cram *)buf;
+	u32 cmd_words[CMD_WDS + EXT_WDS];
 
 	if (off >= XTG_COMMAND_RAM_SIZE) {
 		pr_err("Requested Write len exceeds 8K CRAM size\n");
@@ -1286,8 +1288,6 @@ static ssize_t xtg_cram_write(struct file *filp, struct kobject *kobj,
 
 	/* Program each command */
 	if (count == sizeof(struct xtg_cram)) {
-		struct xtg_cram *cmdp = (struct xtg_cram *)buf;
-		u32 cmd_words[CMD_WDS + EXT_WDS];
 
 		if (!cmdp)
 			return -EINVAL;
-- 
2.31.1

